apps/web/composables/useModernImage/__tests__/useModernImage.spec.ts

```diff
@@ -0,0 +1,123 @@
+// write a test for the addModernImageExtension function
+
+import { mockNuxtImport } from "@nuxt/test-utils/runtime";
+import { useModernImage } from "../useModernImage";
+
+describe('useModernImage', () => {
+
+    const { useRuntimeConfig } = vi.hoisted(() => {
+        return {
+            useRuntimeConfig: vi.fn().mockImplementation(() => {
+                return {
+                    public: {
+                        useAvif: true,
+                        useWebp: true
+                    }
+                }
+            })
+        }
+    });
+
+    mockNuxtImport('useRuntimeConfig', () => {
+        return useRuntimeConfig
+    });
+
+    afterEach(() => {
+        vi.clearAllMocks();
+        useRuntimeConfig.mockImplementation(() => {
+            return {
+                public: {
+                    useAvif: true,
+                    useWebp: true
+                }
+            }
+        });
+    });
+
+    it('should add the avif extension to the url', () => {
+        const { addModernImageExtension } = useModernImage();
+        const url = 'https://example.com/item/images/image.jpg';
+        const res = addModernImageExtension(url);
+        expect(res).toBe('https://example.com/item/images/image.jpg.avif');
+    });
+
+    it('should add the webp extension to the url', () => {
+
+        useRuntimeConfig.mockImplementation(() => {
+            return {
+                public: {
+                    useAvif: false,
+                    useWebp: true
+                }
+            }
+        });
+
+        const { addModernImageExtension } = useModernImage();
+        const url = 'https://example.com/item/images/image.jpg';
+        const res = addModernImageExtension(url);
+        expect(res).toBe('https://example.com/item/images/image.jpg.webp');
+    });
+
+    it('should not add the webp extension to the url if the url has no /item/images/ in it', () => {
+
+        useRuntimeConfig.mockImplementation(() => {
+            return {
+                public: {
+                    useAvif: false,
+                    useWebp: true
+                }
+            }
+        });
+
+        const { addModernImageExtension } = useModernImage();
+        const url = 'https://example.com/images/image.jpg';
+        const res = addModernImageExtension(url);
+        expect(res).toBe('https://example.com/images/image.jpg');
+    });
+
+    it('should not add the webp extension to the url if the base extension is already webp', () => {
+
+        useRuntimeConfig.mockImplementation(() => {
+            return {
+                public: {
+                    useAvif: false,
+                    useWebp: true
+                }
+            }
+        });
+
+        const { addModernImageExtension } = useModernImage();
+        const url = 'https://example.com/item/images/image.webp';
+        const res = addModernImageExtension(url);
+        expect(res).toBe('https://example.com/item/images/image.webp');
+    });
```

Comment:
**suggestion**: Use different `describe` statements for various browser support scenarios. This should make re-using mocks easier.

Reply:
done